### PR TITLE
Add ChainedDelegateConsentService with pluggable vote strategy

### PR DESCRIPTION
**Pull request by michaelabuckley**
Add plumbing for combining IConsentServices with different vote tally strategies

### CODE REVIEW

File: hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/consent/ChainedDelegateConsentServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+import static ca.uhn.fhir.rest.server.interceptor.consent.ChainedDelegateConsentService.withParallelVoting;
+import static ca.uhn.fhir.rest.server.interceptor.consent.ChainedDelegateConsentService.withSerialVoting;
+import static ca.uhn.fhir.rest.server.interceptor.consent.IConsentVoteTest.PARALLEL_STREAM_EXPECTATION;
+import static ca.uhn.fhir.rest.server.interceptor.consent.IConsentVoteTest.SERIAL_STREAM_EXPECTATION;
+import static ca.uhn.fhir.rest.server.interceptor.consent.IConsentVoteTest.splitEnumsToStream;
+import static org.apache.commons.lang3.StringUtils.defaultString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ChainedDelegateConsentServiceTest {
```
Comment (nathandoef) [inline hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/consent/ChainedDelegateConsentServiceTest.java:21]: Great tests!

**Review by nathandoef — APPROVED** [review]
This addition makes things super clean :)

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnum.java [hunk 2]
```diff
@@ -59,53 +59,52 @@ int getPrecedence() {
...
 		}
 	}
+
 	/**
-	 * Evaluate verdicts in order, taking the first "decision" (i.e. first non-PROCEED) verdict.
-	 *
-	 * @return the first decisive verdict, or PROCEED when empty or all PROCEED.
+	 * Does this vote abstain from the verdict?
+	 * I.e. this == PROCEED
+	 * @return false if this vote can be ignored
```
[the remark this answers is not in the document]
Comment (michaelabuckley) [reply hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentOperationStatusEnum.java:66]: oops.  Thanks.

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ChainedDelegateConsentService.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+				.map(nextDelegate -> nextDelegate.startOperation(theRequestDetails, theContextServices)));
+	}
+
+	@Override
+	public boolean shouldProcessCanSeeResource(
+			RequestDetails theRequestDetails, IConsentContextServices theContextServices) {
+		return myDelegates.stream()
+				.map(nextDelegate -> nextDelegate.shouldProcessCanSeeResource(theRequestDetails, theContextServices))
+				.filter(nextShould -> nextShould)
+				.findFirst()
```
[the remark this answers is not in the document]
Comment (michaelabuckley) [reply hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ChainedDelegateConsentService.java:54]: No. We return true of any delegate returns true so they can vote during canSeeResource.

**Pull request merged**